pithos/plugins/notification_icon.py

```diff
@@ -91,7 +111,18 @@ def on_has_name_owner(bus, result):
                 self.prepare_complete()
             else:
                 # No fancy tray here
-                self.prepare_complete()
+                logging.info('Checking if org.gnome.Shell is owned and version')
+                Gio.DBusProxy.new(
+                    self.bus,
+                    Gio.DBusProxyFlags.NONE,
+                    None,
+                    'org.gnome.Shell',
```

Comment:
Just google 'XDG_CURRENT_DESKTOP' and you'll see a billion bug reports of apps using that to sniff the DE and failing.